src/IO/WriteBufferFromS3.h

```diff
@@ -59,7 +60,24 @@ class WriteBufferFromS3 final : public WriteBufferFromFileBase
     String getVerboseLogDetails() const;
     String getShortLogDetails() const;
 
-    struct PartData;
+    struct PartData
+    {
+        Memory<> memory;
+        size_t data_size = 0;
+
+        std::shared_ptr<std::iostream> createAwsBuffer()
```

Comment:
It's not a lot of code but pls check if the implementation l. 70-72 could be moved into the source file.

Reply:
It was complaining about using an incomplete type, so I moved it out from .cpp; I'll recheck.